e2e/nx-plugin-e2e/tests/configuration.e2e.test.ts

```diff
@@ -1,35 +1,138 @@
+  });
+
+  it('should inform about dry run', async () => {
+    const cwd = join(baseDir, 'configure');
+    await materializeTree(tree, cwd);
+
+    const { stderr } = await executeConfigurationGenerator(
+      [project, '--dryRun'],
+      cwd,
+    );
+
+    const cleanedStderr = removeColorCodes(stderr);
+    expect(cleanedStderr).toContain(
+      'NOTE: The "dryRun" flag means no changes were made.',
+    );
+  });
+
+  it('should generate conde-pushup.config.ts file and add target to project.json', async () => {
+    const cwd = join(baseDir, 'configure');
+    await materializeTree(tree, cwd);
+
+    const { code, stdout, stderr } = await executeConfigurationGenerator(
+      [project, '--targetName code-pushup', '--dryRun'],
+      cwd,
+    );
 
     const cleanedStderr = removeColorCodes(stderr);
     expect(code).toBe(0);
 
-    expect(cleanedStderr).toContain(
+    expect(cleanedStderr).not.toContain(
       `NOTE: No config file created as code-pushup.config.js file already exists.`,
     );
+
+    const cleanedStdout = removeColorCodes(stdout);
+
+    expect(cleanedStdout).toContain(generatorExecMsgRegex(cwd));
+    expect(cleanedStdout).toMatch(createConfigMsgRegex);
+    expect(cleanedStdout).toMatch(updateProjectMsgRegex);
+  });
+
+  it('should NOT conde-pushup.config.ts file if one already exists', async () => {
+    const cwd = join(baseDir, 'configure-config-existing');
+    generateCodePushupConfig(tree, projectRoot);
+    await materializeTree(tree, cwd);
+
+    const { code, stdout, stderr } = await executeConfigurationGenerator(
+      [project, '--dryRun'],
+      cwd,
+    );
+
+    const cleanedStderr = removeColorCodes(stderr);
+    expect(code).toBe(0);
+
     expect(cleanedStderr).toContain(
-      'NOTE: The "dryRun" flag means no changes were made.',
+      `NOTE: No config file created as code-pushup.config.ts file already exists.`,
+    );
+
+    const cleanedStdout = removeColorCodes(stdout);
+    expect(cleanedStdout).toContain(generatorExecMsgRegex(cwd));
+    expect(cleanedStdout).not.toMatch(createConfigMsgRegex);
+    expect(cleanedStdout).toMatch(updateProjectMsgRegex);
+  });
+
+  it('should NOT create conde-pushup.config.ts file if skipConfig is given', async () => {
+    const cwd = join(baseDir, 'configure-skip-config');
+    await materializeTree(tree, cwd);
+
+    const { code, stdout } = await executeConfigurationGenerator(
+      [project, '--skipConfig', '--dryRun'],
+      cwd,
     );
 
+    expect(code).toBe(0);
+
     const cleanedStdout = removeColorCodes(stdout);
-    expect(cleanedStdout).toContain(
-      `NX  Generating @code-pushup/nx-plugin:configuration`,
+
+    expect(cleanedStdout).toContain(generatorExecMsgRegex(cwd));
+    expect(cleanedStdout).not.toMatch(createConfigMsgRegex);
+    expect(cleanedStdout).toMatch(updateProjectMsgRegex);
+  });
+
+  it('should NOT add target to project.json if skipTarget is given', async () => {
```

Comment:
How are you testing that the target wasn't added? There's assertion for that in the test.

Reply:
As we use dryRun i test it over the comments as all other checks in this file.

`expect(cleanedStdout).not.toMatch(/^UPDATE.*project.json/m);`

If the package would have been added I would see this msg